datajunction-server/datajunction_server/construction/build.py

```diff
@@ -270,14 +272,22 @@ def join_tables_for_dimensions(
         # onto each select clause
         for select in selects_map:
             if dim_node not in initial_nodes:  # need to join dimension
-                join_asts = _build_joins_for_dimension(
-                    session,
-                    dim_node,
-                    initial_nodes,
-                    tables,
-                    build_criteria,
-                    required_dimension_columns,
-                )
+                if dim_node in dimension_links and dimension_links[dim_node]:
+                    join_query = parse(f"SELECT 1 FROM {dimension_links[dim_node]}")
```

Comment:
This keeps coming up. It shouldn't be too difficult to have dedicated parsers for parts of queries at least the very high-level nodes. Is it worth it though

Reply:
Hmm, I think it's probably not needed for the time being, since there's a pretty easy workaround.